dts/bindings/device_node.yaml.template

```diff
@@ -1,88 +1,149 @@
-title: <short description of the node>
+# Overview of concepts (template below)
```

Comment:
Could we include this documentation in `.rst` format as part of the compiled doc? I think it would probably be better than having it here in  the template

Reply:
Thought the same, but got lazy while trying to figure out where to put it.

Could look at it tomorrow.

Reply:
Moved the overview to the device tree page.

Reply:
@carlescufi ping